### PR TITLE
drivers: ethernet: eth_stm32_hal: Fix cppcheck warning 

### PR DESCRIPTION
**Pull request by spe-ciellt**
Trivial fix for https://github.com/zephyrproject-rtos/zephyr/issues/57336

### CODE REVIEW

**Review by erwango — DISMISSED** [review]
Thanks for this fix

**Review by gautierg-st — DISMISSED** [review]

**Comment by erwango** [comment]
@spe-ciellt Please fix compliance checks reported here: https://github.com/zephyrproject-rtos/zephyr/actions/runs/4872483014/jobs/8691120475?pr=57519#step:9:24

File: drivers/ethernet/eth_stm32_hal.c [hunk 1]
```diff
@@ -1350,7 +1350,7 @@ static uint32_t reverse(uint32_t val)
 	int i;
 
 	for (i = 0; i < 32; i++) {
-		if (val & (1 << i)) {
+		if (val & (1UL << i)) {
```
Comment (GeorgeCGV) [inline drivers/ethernet/eth_stm32_hal.c:1353]: Could be done using existing utils `BIT` define.

```suggestion
		if (val & BIT(n)) {
```

Comment (spe-ciellt) [reply drivers/ethernet/eth_stm32_hal.c]: Updated in new version.

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**